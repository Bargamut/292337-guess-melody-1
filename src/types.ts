export enum Type {
  ARTIST = 'artist',
  GENRE = 'genre'
}

export interface Song {
  artist: string,
  src: string
}

export interface AnswerArtist {
  artist: string,
  picture: string
}

export interface AnswerGenre {
  genre: string,
  src: string
}

export interface QuestionArtist {
  answers: AnswerArtist[],
  song: Song,
  type: Type
}

export interface QuestionGenre {
  answers: AnswerGenre[],
  genre: string,
  type: Type
}
