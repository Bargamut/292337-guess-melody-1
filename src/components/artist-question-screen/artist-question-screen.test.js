import React from 'react';
import renderer from 'react-test-renderer';
import ArtistQuestionScreen from './artist-question-screen.jsx';

const mock = {
  question: {
    type: `artist`,
    song: {
      artist: `Пелагея`,
      src: ``
    },
    answers: [
      {
        artist: `Пелагея`,
        photo: ``
      },
      {
        artist: `Краснознаменная дивизия имени моей бабушки`,
        photo: ``
      },
      {
        artist: `Lorde`,
        photo: ``
      }
    ]
  },
  handleSubmit: jest.fn()
};

it(`Artist Question Screen renders correctly`, () => {
  const {
    question,
    handleSubmit
  } = mock;

  const artistQuestionScreen = renderer.create(
      <ArtistQuestionScreen
        question={question}
        onAnswer={handleSubmit}
        renderAnswer={jest.fn()}
      />,
      {
        createNodeMock: (element) => {
          if (element.type === `audio`) {
            return {
              src: ``
            };
          }

          return null;
        }
      }
  ).toJSON();

  expect(artistQuestionScreen).toMatchSnapshot();
});
