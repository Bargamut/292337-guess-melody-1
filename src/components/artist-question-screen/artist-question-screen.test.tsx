import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Type } from '../../types';
import ArtistQuestionScreen from './artist-question-screen';

const mock = {
  question: {
    type: Type.ARTIST,
    song: {
      artist: `Пелагея`,
      src: ``
    },
    answers: [
      {
        artist: `Пелагея`,
        picture: ``
      },
      {
        artist: `Краснознаменная дивизия имени моей бабушки`,
        picture: ``
      },
      {
        artist: `Lorde`,
        picture: ``
      }
    ]
  },
  handleSubmit: jest.fn()
};

it(`Artist Question Screen renders correctly`, () => {
  const {
    question,
    handleSubmit
  } = mock;

  const artistQuestionScreen = renderer.create(
      <ArtistQuestionScreen
        question={question}
        onAnswer={handleSubmit}
        renderAnswer={jest.fn()}
      />,
      {
        createNodeMock: (element) => {
          if (element.type === `audio`) {
            return {
              src: ``
            };
          }

          return null;
        }
      }
  ).toJSON();

  expect(artistQuestionScreen).toMatchSnapshot();
});
