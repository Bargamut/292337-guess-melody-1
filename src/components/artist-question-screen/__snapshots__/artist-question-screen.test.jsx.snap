// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist Question Screen renders correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <button
      className="track__button track__button--play"
      type="button"
    />
    <audio
      src=""
    />
  </div>
  <form
    className="game__artist"
    onChange={[Function]}
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        type="radio"
        value="answer-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="Пелагея"
          className="artist__picture"
          src=""
        />
        Пелагея
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        type="radio"
        value="answer-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Краснознаменная дивизия имени моей бабушки"
          className="artist__picture"
          src=""
        />
        Краснознаменная дивизия имени моей бабушки
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        type="radio"
        value="answer-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Lorde"
          className="artist__picture"
          src=""
        />
        Lorde
      </label>
    </div>
  </form>
</section>
`;
